src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts

```diff
@@ -160,6 +174,9 @@ export class GKEClusterSettingsComponent
     this.form.valueChanges.pipe(takeUntil(this._unsubscribe)).subscribe(_ => {
       this._updateExternalClusterModel();
       this._updateExternalMachineDeployment();
+      if (this.controlValue(Controls.Zone)?.main) {
```

Comment:
Move this outside to add a subscription for ONLY in Zone change

Reason it will run on every control irrespective of zone is changed or not we should only trigger on Zone change only